styx-api-service/src/main/java/com/spotify/styx/StyxApi.java

```diff
@@ -181,9 +178,6 @@ public void create(Environment environment) {
     final Config config = environment.config();
     final String schedulerServiceBaseUrl = get(config, config::getString, SCHEDULER_SERVICE_BASE_URL)
         .orElse(DEFAULT_SCHEDULER_SERVICE_BASE_URL);
-    final Duration runningStateTtl = get(config, config::getString, STYX_RUNNING_STATE_TTL_CONFIG)
```

Comment:
I moved this to ExtendedWorkflowValidator as it is used by both scheduler and api but seems they set different default values if the configuration is not present

Reply:
This feels like an unclean refactoring, maybe we should align how it works between Api and Scheduler:

```java
final Config staleStateTtlConfig = config.getConfig(STYX_STALE_STATE_TTL_CONFIG);
final TimeoutConfig timeoutConfig = TimeoutConfig.createFromConfig(staleStateTtlConfig);
```

Reply:
hmm yes, how would you prefer it?

Reply:
I'm thinking maybe it's better we do not use `styx.stale-state-ttls` group because it maps to real state 1:1. and `running_max` is not a real state. Maybe it is clearer to introduce `styx.max-running-timeout` as a separated config item, read it in both api and scheduler with default value to `styx.stale-state-ttls.running`?

Reply:
👍

Reply:
fixed